src/Compilers/VisualBasic/Portable/Lowering/LocalRewriter/LocalRewriter_Query.vb

```diff
@@ -61,81 +90,39 @@ Namespace Microsoft.CodeAnalysis.VisualBasic
                     Dim paramRef As New BoundParameter(node.Syntax,
                                                        parameter,
                                                        False,
-                                                       parameter.Type)
+                                                       parameter.Type,
+                                                       hasErrors:=parameter.Type.IsErrorType())
 
-                    If isReservedName AndAlso Not String.Equals(parameterName, StringConstants.Group, StringComparison.Ordinal) Then
+                    If Not paramRef.HasErrors AndAlso isReservedName AndAlso Not String.Equals(parameterName, StringConstants.Group, StringComparison.Ordinal) Then
```

Comment:
>Not paramRef.HasErrors AndAlso [](start = 23, length = 30)

Check that type is not an error type instead?

Reply:
I am not sure this is the right way to handle error recovery. It feels wrong to always end up in a simple case because of this condition. I would like to better understand the scenario you are trying to guard against.

---
In reply to: [137416036](https://github.com/dotnet/roslyn/pull/21356#discussion_r137416036) [](ancestors = 137416036)

Reply:
So for cases where parameter type is an error type, existing code enters the if statment and PopulateRangeVariableMapForAnonymousType attempts to cast the type symbol to AnonymousTypeManager.AnonymousTypePublicSymbol and fails. This new condition ensures that instead with go to the else statement and simply add the bound paramRef to the map.

---
In reply to: [137416477](https://github.com/dotnet/roslyn/pull/21356#discussion_r137416477) [](ancestors = 137416477,137416036)

Reply:
> and simply add the bound paramRef to the map.

Why is this the right thing to do? Let's take a close look at the specific scenario offline.